src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -132,4 +137,20 @@ private WorkspaceEdit GenerateMethodInCodeBlock(RazorCodeDocument code, Generate
 
         return new WorkspaceEdit() { DocumentChanges = new[] { razorTextDocEdit } };
     }
+
+    private static string PopulateMethodSignature(VersionedDocumentContext documentContext, GenerateMethodCodeActionParams actionParams)
+    {
+        var templateWithMethodSignature = s_generateMethodTemplate.Replace(s_methodName, actionParams.MethodName);
+
+        var returnType = actionParams.IsAsync ? "async System.Threading.Tasks.Task" : "void";
```

Comment:
Future PR question: You might have addressed this already, but are there any thoughts on eventually detecting whether a user has a pre-existing using statement so that fully qualifying `Task` wouldn't be necessary?

Reply:
Yep, this will be the next step :) I will be working on exploring a way we can simplify all of the fully qualified types based on using statements that are present.